weed/storage/needle_map/memdb.go

```diff
@@ -61,17 +62,47 @@ func (cm *MemDb) Get(key NeedleId) (*NeedleValue, bool) {
 }
 
 // Visit visits all entries or stop if any error when visiting
+func doVisit(iter iterator.Iterator, visit func(NeedleValue) error) (ret error) {
+	key := BytesToNeedleId(iter.Key())
+	data := iter.Value()
+	offset := BytesToOffset(data[0:OffsetSize])
+	size := BytesToSize(data[OffsetSize : OffsetSize+SizeSize])
+
+	needle := NeedleValue{Key: key, Offset: offset, Size: size}
+	ret = visit(needle)
+	if ret != nil {
+		return
+	}
+	return nil
+}
+
 func (cm *MemDb) AscendingVisit(visit func(NeedleValue) error) (ret error) {
 	iter := cm.db.NewIterator(nil, nil)
+	if iter.First() {
+		if ret = doVisit(iter, visit); ret != nil {
+			return
+		}
+	}
 	for iter.Next() {
-		key := BytesToNeedleId(iter.Key())
-		data := iter.Value()
-		offset := BytesToOffset(data[0:OffsetSize])
-		size := BytesToSize(data[OffsetSize : OffsetSize+SizeSize])
-
-		needle := NeedleValue{Key: key, Offset: offset, Size: size}
-		ret = visit(needle)
-		if ret != nil {
+		if ret = doVisit(iter, visit); ret != nil {
+			return
+		}
+	}
+	iter.Release()
+	ret = iter.Error()
+
+	return
+}
+
+func (cm *MemDb) DescendingVisit(visit func(NeedleValue) error) (ret error) {
```

Comment:
this memdb change seems an unrelated change.

this `DescendingVisit` is not used.

Reply:
The change is related and this function is used to check the meta data of only chunks from the end

Reply:
I could not find the usage of this function.

Reply:
> I could not find the usage of this function.

![image](https://user-images.githubusercontent.com/9497591/197468905-c798b200-5cec-4e50-ae67-79cf7121d1ea.png)